sdk/framework/backend.go

```diff
@@ -108,6 +112,18 @@ type CleanupFunc func(context.Context)
 // InvalidateFunc is the callback for backend key invalidation.
 type InvalidateFunc func(context.Context, string)
 
+// InitializeFunc is the callback, which if set, will be invoked via
+// Initialize() just after a plugin has been mounted.
+type InitializeFunc func(context.Context, logical.Storage) error
```

Comment:
Wondering if we want the backends to receive an `InitializationRequest` object or not. If we end up extending that object to include more information it will future proof the interface a bit